torch/autograd/variable.py

```diff
@@ -122,8 +122,11 @@ def __repr__(self):
     def __bool__(self):
         if self.data.numel() == 0:
             return False
-        raise RuntimeError("bool value of Variable objects containing non-empty " +
-                           torch.typename(self.data) + " is ambiguous")
+        elif self.numel() == 1:
+            return torch.squeeze(self.data)[0] != 0
```

Comment:
I think it's better to not repeat yourself and to do `self.data.__bool__()`

Reply:
Agreed, thanks.